File: 1213-intersection-of-three-sorted-arrays/1213-intersection-of-three-sorted-arrays.cpp
class Solution {
public:
    vector<int> arraysIntersection(vector<int>& a, vector<int>& b, vector<int>& c) {
        int p = 0, q = 0, r = 0;
        vector<int> ans;
        while (p < a.size() and q < b.size() and r < c.size()) {
            if (a[p] == b[q] and b[q] == c[r]) {
                ans.push_back(a[p]);
                ++p; ++q; ++r;
            } else {
                if (a[p] < b[q]) ++p;
                else if (b[q] < c[r]) ++q;
                else ++r;
            }
        }
        return ans;
    }
};

File: 1213-intersection-of-three-sorted-arrays/1213-intersection-of-three-sorted-arrays.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @param {number[]} arr3
 * @return {number[]}
 */
var arraysIntersection = function (arr1, arr2, arr3) {
    let p = 0,
        q = 0,
        r = 0;
    var ans = [];
    while (p < arr1.length && q < arr2.length && r < arr3.length) {
        if (arr1[p] == arr2[q] && arr2[q] == arr3[r]) {
            ans.push(arr1[p]);
            ++p;
            ++q;
            ++r;
        } else {
            if (arr1[p] < arr2[q]) ++p;
            else if (arr2[q] < arr3[r]) ++q;
            else ++r;
        }
    }
    return ans;
};
